crates/edr_evm/src/state/remote/cached.rs

```diff
@@ -118,6 +104,30 @@ impl State for CachedRemoteState {
     }
 }
 
+/// Fetches an account from the remote state. If it exists, code is split off
+/// and stored separately in the provided cache.
+fn fetch_remote_account(
```

Comment:
Out of curiosity, why make this a separate function instead of making it a method of `CachedRemoteState`?